File: src/test/java/workflowsuite/kpi/client/time/ServerTimeProviderTest.java
package workflowsuite.kpi.client.time;

import workflowsuite.kpi.client.settings.ConfigurationProvider;

import javax.net.SocketFactory;
import java.io.IOException;
import java.net.URI;
import java.net.URISyntaxException;

public final class ServerTimeProviderTest {
    //@Test
    public void getNtpData() throws URISyntaxException, IOException {
        TimeServerConfiguration configuration = new TimeServerConfiguration();
        configuration.setEndpoint(URI.create("tcp://msk-tfs-lab09:9701/"));
        ServerTimeProvider timeProvider = new ServerTimeProvider(ConfigurationProvider.wrap(configuration), SocketFactory.getDefault());
        NtpData ntpData = timeProvider.getNtpData();
    }
}

File: test/workflowSuite/kpiClient/time/ServerTimeProviderTest.kt
package workflowSuite.kpiClient.time

import org.junit.Test
import workflowSuite.kpiClient.settings.SimpleConfigurationProvider
import java.net.URI

class ServerTimeProviderTest {
    @Test
    fun getNtpData() {
        val configuration = SimpleConfigurationProvider(TimeServerConfiguration(URI("tcp://msk-dev-foris:9701/")))
        val timeProvider = ServerTimeProvider(configuration)
        val ntpData = timeProvider.GetNtpData()
    }

}